data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -575,6 +580,18 @@ public Builder<T> withTimeReceived(final Instant timeReceived) {
             return this;
         }
 
+        /**
+         * Sets the event handle
+         *
+         * @param eventHandle event handle
+         * @return returns the builder
+         * @since 2.10
+         */
+        public Builder<T> withEventHandle(final EventHandle eventHandle) {
```

Comment:
This needs to be part of `BaseEventBuilder`. I'm also tempted to say that we only support it in there by making this particular function package protected, but that may also be a bit of work to accomplish.

Reply:
Actually, as this is going to be used primarily by the `aggregate` processor, I think it would be reasonable to expose this only in `BaseEventBuilder`. You won't have to change every source. Just change that one processor.

Reply:
Do you want me to make this change to use BaseEventBuilder in aggregate processor now or can it be part of a future PR?

Reply:
Let's add `BaseEventBuilder` in this one.

Reply:
Or you can make this package protected for this PR. Either way.